### PR TITLE
Add Flexible Joint States message design doc

### PR DESCRIPTION
**Pull request by bmagyar**
Draft for flexible joint states message

### CODE REVIEW

**Comment by bmagyar** [comment]
@jordan-palacios @Karsten1987 I've tried to collect my current ideas and a general specification of what I'd like to see out of this. Please scrutinize and suggest changes :)

**Comment by gavanderhoorn** [comment]
Reminded me of https://github.com/ros-controls/ros_controllers/issues/416 (there are probably others).

**Comment by bmagyar** [comment]
Thanks for that ref @gavanderhoorn ! The end of the unholy trinity of pos-vel-eff is drawing nigh!

**Comment by gavanderhoorn** [comment]
Would it make sense to try and get some insight into how other control frameworks are modelling / encoding this type of data?

Might provide valuable insights.

What about going for the channel approach (ie: give each `value-type` its own topic)? `JointState` is right now typically used as a bus. I appreciate the added complexity of having `N` topics for `N` `value-type`s, but it might be worth it.

File: design_drafts/flexible_joint_states_msg.md [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+Assumptions in the current setup:
+* the values in `position`, `velocity` and `effort` are aligned with the joint names in `name`.
+
+# Goal
+
+To define a new message that
+1. allows to report additional values on top of position, velocity and effort,
+2. does not enforce defining position, velocity and effort for every joint,
+3. does not enforce defining values for every joint in every value-type,
```
Comment (piraka9011) [inline design_drafts/flexible_joint_states_msg.md]: Does this mean specifying a default somewhere instead?

File: design_drafts/flexible_joint_states_msg.md [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+string[] name
+InterfaceValue[] interface_value
+  string interface_name
+  float64[] value
+```
+
+where `name` stands for a list of joint names, `interface_value` is a vector of a new message, holding the name of the interface and a list of values.
+Assumptions:
+* `interface_name` needs to be kept standard somehow
```
Comment (piraka9011) [inline design_drafts/flexible_joint_states_msg.md]: Is this necessary? Maybe in a higher layer present the user with a warning or use a config file/URDF that all nodes can read from.

File: design_drafts/flexible_joint_states_msg.md (hunk 1 above)
Comment (bmagyar) [reply design_drafts/flexible_joint_states_msg.md]: With the current solution, you either leave the vector empty or use extremal values of your preference. In an ideal solution the vector wouldn't even exist

File: design_drafts/flexible_joint_states_msg.md (hunk 2 above)
Comment (bmagyar) [reply design_drafts/flexible_joint_states_msg.md]: I should have added an example here probably. These interfaces are basically the same as what transmission parsing reads, for isntance [here](https://github.com/ros-controls/ros_control/blob/melodic-devel/transmission_interface/test/urdf/simple_transmission_loader_full.urdf#L9). 
This assumes that some form of `hardware_interface/PositionJointInterface` is kept standard throughout.

Please ask away and request additions, this is purely a draft doc

**Comment by bmagyar** [comment]
This document is by no means done, please feel free to open further PR-s with more proposals.

**Pull request merged**